enterprise_access/apps/content_assignments/api.py

```diff
@@ -272,49 +277,33 @@ def allocate_assignments(assignment_configuration, learner_emails, content_key,
     # content_price_cents, and then persist that in the assignment records.
     content_quantity = content_price_cents * -1
 
-    # Fetch any existing assignments for all pairs of (learner, content) in this assignment config.
-    existing_assignments = get_assignments_for_admin(
+    lms_user_ids_by_email, emails_by_lms_user_id = _map_allocation_emails_with_lms_user_ids(
+        learner_emails_to_allocate,
+    )
```

Comment:
One of the big changes here - we now fetch mappings between email <-> lms_user_id up-front, and then use those mappings as needed during assignment re-allocation and creation.